src/sentry/issues/occurrence_consumer.py

```diff
@@ -42,6 +46,31 @@ class EventLookupError(Exception):
     pass
 
 
+def create_rate_limit_key(fingerprint: str) -> str:
+    rate_limit_key = f"occurrence_rate_limit:{fingerprint}"
+    return rate_limit_key
+
+
+def is_rate_limited(
+    fingerprint: str,
+) -> bool:
+    rate_limit_enabled = options.get("issues.occurrence-consumer.rate-limit.enabled")
+    if rate_limit_enabled:
```

Comment:
nit: you could change this to a guard clause: https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html
```if not rate_limit enabled:
    return False
```
... rest of the code, to reduce nesting.